### PR TITLE
fix: (Platform) i18n remove select, plurals

### LINKED ISSUE #5098: Remove i18n "select" and "plural" usage in flavour of something more straight forward
#### Is this a bug, enhancement, or feature request?
SAP team uses internaly i18n plugin which does not support  i18n "select" and "plural" syntax

### PR DESCRIPTION
**Pull request by platon-rov**
#### Please provide a link to the associated issue.

Closes #5098.

#### Please provide a brief summary of this pull request.

Code of the Platform components refactored to get rid of plural & select features of the i18n. The [wiki page](https://github.com/SAP/fundamental-ngx/wiki/Internationalization-Supporting-in-@fundamental-ngx-platform#select--plural) about internationalization updated.

As i18n isn't an Angular directive - we're highly limited in next things:

* Cannot **bind** i18n id - so we cannot implement & use helper component.
* Cannot set i18n markers **inside** of the loops, components code (there exist hacky solution but it isn't applicable as translations extracting isn't working).

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the [PR Review Checklist](https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist) are addressed.
- [X] Run npm run build-pack-library and test in external application

Documentation checklist:
- [X] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

File: libs/platform/src/lib/components/value-help-dialog/models/vhd-strategy.enum.ts [hunk 1]
```diff
@@ -1,17 +1,3 @@
-export enum VhdDefineStrategy {
```
Comment (platon-rov) [inline libs/platform/src/lib/components/value-help-dialog/models/vhd-strategy.enum.ts:1]: @mykolavorobiov i see that you edited that file a week ago, I found an unneeded interface, removed it, but maybe it had the implicit sence, please take a look.

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html [hunk 2]
```diff
@@ -483,81 +483,151 @@ <h3 [innerText]="noDataText"></h3>
...
+            *ngIf="messageStripType === 'error'; else filesAndFoldersMovedTo"
+            i18n="@@platformUploadCollection.Message.Move.FoldersAndFiles.Failed"
+        >
+            {{ count.folder }} folders and {{ count.file}} files have been failed to move to
+        </ng-container>
+        <ng-template #filesAndFoldersMovedTo i18n="@@platformUploadCollection.Message.Move.FoldersAndFiles">
+            {{ count.folder }} folders and {{ count.file}} files have been moved to
+        </ng-template>
+        
+        {{ payload.to ? payload.to.name : 'All files' }}.
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html:585]: `All files` is out of i18n marker

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html [hunk 3]
```diff
@@ -483,81 +483,151 @@ <h3 [innerText]="noDataText"></h3>
...
+            *ngIf="messageStripType === 'error'; else foldersMovedTo"
+            i18n="@@platformUploadCollection.Message.Move.Folders.Failed"
+        >
+            {{ count.folder }} folders have been failed to move to
+        </ng-container>
+        <ng-template #foldersMovedTo i18n="@@platformUploadCollection.Message.Move.Folders">
+            {{ count.folder }} folders have been moved to
+        </ng-template>
+        
         {{ payload.to ? payload.to.name : 'All files' }}.
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html:601]: `All files` is not marked for translation

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html [hunk 4]
```diff
@@ -483,81 +483,151 @@ <h3 [innerText]="noDataText"></h3>
...
+            *ngIf="messageStripType === 'error'; else filesMovedTo"
+            i18n="@@platformUploadCollection.Message.Move.Files.Failed"
+        >
+            {{ count.file }} files have been failed to move to
+        </ng-container>
+        <ng-template #filesMovedTo i18n="@@platformUploadCollection.Message.Move.Files">
+            {{ count.file }} files have been moved to
+        </ng-template>
+        
         {{ payload.to ? payload.to.name : 'All files' }}.
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html:617]: out of i18n

File: libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html [hunk 5]
```diff
@@ -483,81 +483,151 @@ <h3 [innerText]="noDataText"></h3>
...
+            *ngIf="messageStripType === 'error'; else oneItemMovedTo"
+            i18n="@@platformUploadCollection.Message.Move.OneItem.Failed"
+        >
+            {{ payload.items[0].name }} have been failed to move to
+        </ng-container>
+        <ng-template #oneItemMovedTo i18n="@@platformUploadCollection.Message.Move.OneItem">
+            {{ payload.items[0].name }} have been moved to
+        </ng-template>
+
         {{ payload.to ? payload.to.name : 'All files' }}.
```
Comment (dimamarksman) [inline libs/platform/src/lib/components/upload-collection/upload-collection/upload-collection.component.html:631]: out of i18n

**Review by dimamarksman — CHANGES_REQUESTED** [review]
LGTM
Just one comment

**Review by dimamarksman — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by olshansky — APPROVED** [review]

**Pull request merged**